test/quantization/test_quantize_fx.py

```diff
@@ -2622,11 +2622,52 @@ def forward(self, input):
             self.checkGraphModeFxOp(
                 M(is_module), data, quant_type, quantized_nodes[is_module])
 
+    def _test_norm_float16_impl(
+            self, float_module, float_op, op_args, data,
+            skip_op_arg_for_functional=False):
+        ''' Test for normalization op, float_op can be torch op or functional op,
+        op_args is a list of positional argument for the module/op
+        '''
+        class M(torch.nn.Module):
+            def __init__(self, is_module):
+                super(M, self).__init__()
+                self.is_module = is_module
+                if self.is_module:
+                    self.op = float_module(*op_args)
+                else:
+                    self.op = float_op
+
+            def forward(self, input):
+                if self.is_module:
+                    return self.op(input)
+                else:
+                    args = [input]
+                    if not skip_op_arg_for_functional:
+                        args += op_args
+                    return self.op(*args)
```

Comment:
optional: this seems hard to read, at least for me (it may be different for others). I wonder if there were two less complicated classes, maybe one for modules and one for functions, if it would be easier to read.  I personally would value readability more than code reuse in this context.  I'm flexible though.

Reply:
sure, this is a little bit hard to read, I can split it to two modules